engines/mock/mockengine.go

```diff
@@ -7,19 +7,22 @@ import (
 
 	"github.com/taskcluster/taskcluster-worker/engines"
 	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	log "github.com/taskcluster/taskcluster-worker/log"
 	"github.com/taskcluster/taskcluster-worker/runtime"
 )
 
 type engine struct {
 	engines.EngineBase
+	Log log.Logger
 }
 
 func init() {
 	// Register the mock engine as an import side-effect
 	extpoints.EngineProviders.Register(func(
-		options extpoints.EngineOptions,
+		options *extpoints.EngineOptions,
 	) (engines.Engine, error) {
-		return engine{}, nil
+		logger := options.Environment.Log.WithField("engine", "mock")
```

Comment:
I for the record: `Log` wasn't added to `EngineOptions` :)


Also I think `.WithField("engine", engineName)` is something we should do when we construct the `EngineOptions`. Because there is one place in the main function where we initialize the engine, that is it... So we won't have to implement this in every engine and we are sure we always write the tag the same way.

Basically, create child loggers, when passing a logger into a new component. Rather than doing it inside the constructor of the component. So if mock engine wanted to pass a logger to some sub-component it uses, it could add tags there.